src/main/java/com/cloudbees/dockerpublish/DockerBuilder.java

```diff
@@ -50,7 +49,8 @@
     * for the actual HTML fragment for the configuration screen.
     */
     @DataBoundConstructor
-    public DockerBuilder(String repoName, String repoTag, boolean skipPush, boolean noCache, boolean forcePull, boolean skipBuild, boolean skipDecorate, boolean skipTagLatest, String dockerfilePath) {
+    public DockerBuilder(DockerRegistryEndpoint registry, String repoName, String repoTag, boolean skipPush, boolean noCache, boolean forcePull, boolean skipBuild, boolean skipDecorate, boolean skipTagLatest, String dockerfilePath) {
```

Comment:
I would recommend putting only truly mandatory parameters with no sensible default in the `@DataBoundConstructor`. All else can get `@DataBoundSetter`s. That makes it much easier to maintain Java-level compatibility later.